src/Agent.Worker/JobRunner.cs

```diff
@@ -218,8 +218,8 @@ public sealed class JobRunner : AgentService, IJobRunner
                 catch (OperationCanceledException ex)
                 {
                     // set the job to canceled
+                    // don't log error issue to job ExecutionContext, since server owns the job level issue
                     Trace.Error($"Caught exception: {ex}");
-                    jobContext.Error(ex);
```

Comment:
My only concern would be that an OperationCanceledException could bubble for some other reason (not initiated by job cancel). I'll defer to you to consider whether my concern is legitimate or practical.

My suggestion would be to log something to the job context. Maybe the message as info, and full exception to debug?

Again, I'll defer to you to consider what's best to do. I'm fine with the PR either way.